### PR TITLE
Integration testing against peridio-cloud

### PR DESCRIPTION
**Pull request by KronicDeth**

### CODE REVIEW

File: .github/workflows/new-build.yml [hunk 1]
```diff
@@ -39,6 +39,21 @@ jobs:
     runs-on: "${{ matrix.os }}"
     env:
       CARGO_NET_GIT_FETCH_WITH_CLI: true
+      PERIDIO_CLOUD_PATH: tests/support/peridio-clou
```
Comment (andresatvery) [inline .github/workflows/new-build.yml:42]: This seems like a typo of  `tests/support/peridio-cloud`?🤔

File: tests/integration_tests.rs [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+    let api_key = user.create_api_key();
+    let ca_path = ca_path_buf.into_os_string().into_string().unwrap();
+
+    PERIDIO_CLOUD_API.init();
+
+    let assert = Command::cargo_bin("peridio-cli")
+        .unwrap()
+        .args(&["--base-url", &base_url])
+        .args(&["--ca-path", &ca_path])
+        .args(&["--api-key", &api_key])
```
Comment (andresatvery) [inline tests/integration_tests.rs:83]: I don't think you need to reference `&` the array/list argument on those 3 places

File: tests/integration_tests.rs [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+        let email = format!("{}@{}", username, email_domain);
+        let password = "peridio-cli";
+        let prn_temp_path = NamedTempFile::new().unwrap().into_temp_path();
+
+        match peridio_cloud_core_mix_command()
+            .arg("peridio.user.create")
+            .args(&["--username", username.as_str()])
+            .args(&["--email", email.as_str()])
+            .args(&["--password", password])
+            .args(&["--prn-file", prn_temp_path.to_str().unwrap()])
```
Comment (andresatvery) [inline tests/integration_tests.rs:233]: Same comment as before ^^/

File: tests/integration_tests.rs [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+
+    fn create_api_key(&self) -> String {
+        let note = random_name();
+        let user_token_temp_path = NamedTempFile::new().unwrap().into_temp_path();
+
+        match peridio_cloud_core_mix_command()
+            .arg("peridio.user_token.create")
+            .args(&["--user-prn-file", self.prn_temp_path.to_str().unwrap()])
+            .args(&["--note", &note])
+            .args(&["--user-token-file", user_token_temp_path.to_str().unwrap()])
```
Comment (andresatvery) [inline tests/integration_tests.rs:263]: Same comment as before ^^/

File: tests/integration_tests.rs [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+        Self {
+            child: Some(Self::child()),
+        }
+    }
+
+    fn child() -> Child {
+        match peridio_cloud_api_mix_command()
+            .stderr(Stdio::inherit())
+            .stdout(Stdio::inherit())
+            .args(&["phx.server"])
```
Comment (andresatvery) [inline tests/integration_tests.rs:333]: Same comment as before ^^/

**Review by andresatvery — APPROVED** [review]
🎉 LGTM! Just a couple small messages, but not blocking!

File: tests/integration_tests.rs (hunk 2 above)
Comment (KronicDeth) [reply tests/integration_tests.rs:83]: It is necessary to change the type from `String` to `&str`
Screenshot 2024-02-27 at 10 17 17 AM

File: tests/integration_tests.rs (hunk 3 above)
Comment (KronicDeth) [reply tests/integration_tests.rs:233]: They need to all be `&str`
Screenshot 2024-02-27 at 10 21 40 AM
Screenshot 2024-02-27 at 10 21 46 AM
Screenshot 2024-02-27 at 10 21 59 AM

File: tests/integration_tests.rs (hunk 4 above)
Comment (KronicDeth) [reply tests/integration_tests.rs:263]: They need to all be `&str`

Screenshot 2024-02-27 at 10 24 33 AM
Screenshot 2024-02-27 at 10 24 39 AM
Screenshot 2024-02-27 at 10 24 51 AM

File: tests/integration_tests.rs (hunk 5 above)
Comment (KronicDeth) [reply tests/integration_tests.rs]: Fixed

File: tests/integration_tests.rs (hunk 2 above)
Comment (KronicDeth) [reply tests/integration_tests.rs:83]: Fixed

File: tests/integration_tests.rs (hunk 3 above)
Comment (KronicDeth) [reply tests/integration_tests.rs:233]: `&` removed from in front of `[`.

File: tests/integration_tests.rs (hunk 4 above)
Comment (KronicDeth) [reply tests/integration_tests.rs:263]: `&` removed from in front of `[`.

**Review by andresatvery — APPROVED** [review]

**Pull request merged**